### DISCUSSION TITLE
i18n sub-path routing - custom parsing of locale in url

### DISCUSSION THREAD

**Discussion by lviit** [vercel/next.js#21871] [Help]
Hello 👋 

When using Internationalized routing, the locale in the url needs to be a  UTS Locale Identifier. Is it possible to somehow handle custom parsing of the locale in the url, for example if I would like to have an url such as: `/foo-en/blog`, I could somehow resolve the locale to `en`?

I know a workaround would be to just use a dynamic route like `/[foo-locale]/blog` instead, but I would like to be able to leverage the other benefits of using the Internationalized Routing feature in Next 10.

The use case is a scenario where I'm migrating from an old system, and would like to keep the url's exactly the same as before.

**Comment by laarakus** [comment]
sub path routing is stopping us from migrating to the new version of next-translate because they use nextjs i18n built in feature,
our url format is like this: /[country]/[language]/... 
like most e commerce websites we use country,language in the url to extract the store id:
/be/fr => store-1020
/be/nl => store-1021
and use the language in the url to feed i18nProvider HOC to translate frontend labels
maybe we can migrate if custom sub path mapper is allowed in next conf file

**Reply by juliuscc** [reply]
We also have this problem

**Comment by juliuscc** [comment]
Just to make it easier for future developers that end up here, I think this discussion is similar to [this](https://github.com/vercel/next.js/discussions/25215) discussion.
